### PR TITLE
Add sourceBuffer.changeType sample

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Hello @wolenetz,

What do you think of this sample?
I'll mention Origin Trial when it's confirmed.

I'll also upload these files to a Google Cloud Storage instance.

### CODE REVIEW

**Comment by wolenetz** [comment]
@beaufortfrancois No Origin Trial was started; we shipped it on-by-default in M70.
Looking at the sample now.

File: media/sourcebuffer-changetype.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+feature_name: Codec and container switching in MSE
+chrome_version: 70
+feature_id: 5719220952236032
+check_min_version: true
+---
+
+<h3>Background</h3>
+<p>
+Chrome is experimenting with adding support for improved cross-codec or
```
Comment (wolenetz) [inline media/sourcebuffer-changetype.html]: s/experimenting with adding/has added/

File: media/sourcebuffer-changetype.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+</p>
+<p>
+This sample demonstrates the use of this new method by appending different
+types of media bytes to a single SourceBuffer of a MediaSource attached to an
+audio element.
+</p>
+
+<audio controls></audio>
+
+{% include output_helper.html initial_output_content=initial_output_content %}
```
Comment (wolenetz) [inline media/sourcebuffer-changetype.html:23]: I'm unclear what this line actually does. rubber-stamping it.

File: media/sourcebuffer-changetype.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+<script>
+  log = ChromeSamples.log;
+</script>
+
+{% include js_snippet.html filename='sourcebuffer-changetype.js' %}
```
Comment (wolenetz) [inline media/sourcebuffer-changetype.html:29]: Ditto

File: media/sourcebuffer-changetype.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+const mediaSource = new MediaSource();
+const audio = document.querySelector('audio');
+audio.src = URL.createObjectURL(mediaSource);
+
+mediaSource.addEventListener('sourceopen', function() {
+  URL.revokeObjectURL(audio.src);
+  feedAudioElement(0 /* index */);
```
Comment (wolenetz) [inline media/sourcebuffer-changetype.js]: Perhaps for readability, s/index/resources_index/ and move the "const resources = ...;" block to before this event listener?

File: media/sourcebuffer-changetype.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  try {
+    let sourceBuffer;
+    const resource = resources[index];
+
+    if (mediaSource.sourceBuffers.length == 0) {
+      log('Adding source buffer with type: ' + resource.type);
+      sourceBuffer = mediaSource.addSourceBuffer(resource.type);
+      sourceBuffer.mode = 'sequence';
```
Comment (wolenetz) [inline media/sourcebuffer-changetype.js]: Perhaps add comment that 'sequence' is used here purely for simplicity, and that a more sophisticated MSE app would know which timestampOffset and appendWindow to use for each appendBuffer operation when splicing media.

File: media/sourcebuffer-changetype.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    log('Appending data to audio element...');
+    sourceBuffer.appendBuffer(data);
+    sourceBuffer.addEventListener('updateend', function() {
+      log('Source buffer updated');
+      feedAudioElement(++index);
+    }, { once: true });
+
+  } catch(error) {
+    log('Argh! ' + error);
+    feedAudioElement(++index);
```
Comment (wolenetz) [inline media/sourcebuffer-changetype.js]: At this point, not much reliability in the outcome of feedAudioElement(++index). Error could have resulted due to lack of support for the mime type (in changeType() or addSourceBuffer().

**Review by wolenetz — CHANGES_REQUESTED** [review]
Looks pretty good. Just a few nits.

File: media/sourcebuffer-changetype.html (hunk 1 above)
Comment (beaufortfrancois) [reply media/sourcebuffer-changetype.html]: Done.

File: media/sourcebuffer-changetype.html (hunk 2 above)
Comment (beaufortfrancois) [reply media/sourcebuffer-changetype.html:23]: This is part of the samples architecture. Please ignore.

File: media/sourcebuffer-changetype.html (hunk 3 above)
Comment (beaufortfrancois) [reply media/sourcebuffer-changetype.html:29]: Ditto

File: media/sourcebuffer-changetype.js (hunk 4 above)
Comment (beaufortfrancois) [reply media/sourcebuffer-changetype.js]: Done.

File: media/sourcebuffer-changetype.js (hunk 5 above)
Comment (beaufortfrancois) [reply media/sourcebuffer-changetype.js]: Done.

File: media/sourcebuffer-changetype.js (hunk 6 above)
Comment (beaufortfrancois) [reply media/sourcebuffer-changetype.js]: I think that's OK.
Goal here is to iterate through buffers and make this demo fail gracefully in Safari (with no support for webm container).

Comment (wolenetz) [reply media/sourcebuffer-changetype.js]: Ack

**Pull request merged**